### PR TITLE
[MOBILE-24] Add deep link event listener

### PR DESCRIPTION
**Pull request by oristanovic**
### What do these changes do?
Add a deep link event listener to the NetStandard lib, mirrors the event listener in Cordova/React Native.

### How did you verify these changes?
Created a xamarin forms app and tested manually on iOS and Android, checked via logs that the event callback is executed with the correct parameters.

### CODE REVIEW

File: src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Abstractions/IAirship.cs [hunk 1]
```diff
@@ -53,5 +53,7 @@ int MessageCenterCount
         Channel.TagGroupsEditor EditChannelTagGroups();
         
         Attributes.AttributeEditor EditAttributes();
+
+        void RegisterEventListener(Action<string, Dictionary<string, string>> onEventReceived);
```
Comment (rlepinski) [inline src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Abstractions/IAirship.cs:57]: C# has an event listener pattern that we should follow:

```
        public delegate void DeepLinkHandler (string deepLink);

        public event DeepLinkHandler OnDeepLink;
``` 

Then you add/remove using:
```
UAirship.instance.OnDeepLink += OnDeepLink;
UAirship.instance.OnDeepLink -= OnDeepLink;
```

Comment (oristanovic) [reply src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Abstractions/IAirship.cs]: Noice, updated the listener.

File: src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Abstractions/Analytics/DeepLinkEvent.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+﻿/*
+ Copyright Airship and Contributors
+*/
+
+using System.Collections.Generic;
+
+namespace UrbanAirship.NETStandard.Analytics
```
Comment (rlepinski) [inline src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Abstractions/Analytics/DeepLinkEvent.cs:7]: do we need this still?

File: src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.iOS/Airship.cs [hunk 3]
```diff
@@ -63,6 +66,22 @@ public string NamedUser
...
+            add
+            {
+                onDeepLinkReceived += value;
+                UAirship.Shared().WeakDeepLinkDelegate = this;
+            }
+
+            remove
+            {
+                onDeepLinkReceived -= value;
+                UAirship.Shared().WeakDeepLinkDelegate = null;
```
Comment (rlepinski) [inline src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.iOS/Airship.cs:81]: do we need to see if the total listeners is empty before removing the delegate?

**Pull request merged**